compiler/rustc_mir/src/transform/lower_slice_len.rs

```diff
@@ -0,0 +1,144 @@
+//! This pass lowers calls to core::slice::len to just Len op.
+//! It should run before inlining!
+
+use crate::transform::MirPass;
+use rustc_hir::def_id::DefId;
+use rustc_middle::mir::visit::MutVisitor;
+use rustc_middle::mir::*;
+use rustc_middle::ty::{self, TyCtxt};
+
+pub struct LowerSliceLenCalls;
+
+impl<'tcx> MirPass<'tcx> for LowerSliceLenCalls {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        lower_slice_len_calls(tcx, body)
+    }
+}
+
+pub fn lower_slice_len_calls<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+    // FIXME: (shamatar) Can be tuned later
+    // if tcx.sess.mir_opt_level() < 3 {
+    //     return;
+    // }
+
+    // Optimization is once per basic block, so we just visit terminators
+    let languate_items = tcx.lang_items();
+    let slice_lang_item_def_id = if let Some(slice_lang_item) = languate_items.slice_impl() {
+        slice_lang_item
+    } else {
+        // there is no language item to compare too :)
+        return;
+    };
+    let len_symbol = rustc_span::Symbol::intern("len");
```

Comment:
This should probably be turned into a statically interned symbol in the `sym` module.

Reply:
If this PR is found useful I can try to add it too